src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,118 @@
 package core.basesyntax;
 
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws DataGettingException {
+
+        if (names == null || data == null || dateFrom == null
+                || dateTo == null) {
+            throw new DataGettingException("one or more input arguments are null");
+        }
+
+        int dayFrom = getNumberOfDay(dateFrom, "day");
+        int monthFrom = getNumberOfDay(dateFrom, "month");
+        int yearFrom = getNumberOfDay(dateFrom, "year");
+
+        int dayTo = getNumberOfDay(dateTo, "day");
+        int monthTo = getNumberOfDay(dateTo, "month");
+        int yearTo = getNumberOfDay(dateTo, "year");
+
+        if (yearFrom > yearTo || (yearFrom == yearTo && monthFrom > monthTo)
+                || (yearFrom == yearTo && monthFrom == monthTo && dayFrom > dayTo)) {
+            throw new DataGettingException("Start date cannot be later than end date");
+        }
+
+        String[] startResult = new String[data.length];
+
+        for (int i = 0; i < data.length; i++) {
+            String[] dataArrayI = data[i].split(" ");
+            String[] dataPeriod = dataArrayI[0].split("\\.");
+
+            int day = Integer.parseInt(dataPeriod[0]);
+            int month = Integer.parseInt(dataPeriod[1]);
+            int year = Integer.parseInt(dataPeriod[2]);
+
+            boolean withinDateRange = (year == yearFrom && (month >= monthFrom && day >= dayFrom)
+                    && (month <= monthTo && day <= dayTo))
+                    || (year == yearTo && (month <= monthTo && day <= dayTo)
+                    && (month >= monthFrom && day >= dayFrom));
+
+            if (withinDateRange) {
+                String name = dataArrayI[1];
+                int hours = Integer.parseInt(dataArrayI[2]);
+                int money = Integer.parseInt(dataArrayI[3]);
+                int salary = hours * money;
+
+                addPersonToStartResult(startResult, name, salary);
```

Comment:
The method `addPersonToStartResult` is called to add or update a person's salary in the result array. Ensure that this method correctly handles cases where a person's name appears multiple times in the data.